### DISCUSSION TITLE
What is the behavior for Rts.reopenCase() method

### DISCUSSION THREAD

**Discussion by c-ankur** [americanexpress/unify-flowret#20] [General] [answered]
Hi,

I was wondering if an already started case (which got some error, let's say) can be reopened. Looking at the source code, there is a method `public ProcessContext reopenCase(String caseId, String ticket, boolean pendBeforeResume, String pendWb)`

- Could you please describe how this method should be used?
- What is the expected values as input, what their meanings are, and are they mandatory?
- Basically, some documentation on how and when this method can/cannot be used?

**Comment by deepakarora3** [comment, marked as answer]
Hi,

Thanks for the query. An important point to note is that reopen case is only for cases which have gone on to their completion / are finalized. In order words the end step has been called on them. If a case is pended somewhere with an error, it is not a candidate for reopening. Having said this, I do realize that there may be a requirement to finalize (end) a case which is stuck somewhere and we don't yet have an API to do that. Let me put that as part of the back log which I will implement shortly.

The way reopen works is that you need to have a ticket defined as part of the process definition that tells Flowret where to resume the case from when it is reopened. The pendBeforeResume tells Flowret if the case is to be pended in a work basket after the reopen is called. If true then the name of the work basket needs to be specified. If false, then after the reopen is done, the case will resume processing.

Hope this clarifies and apologies for the delay in getting back.
